packages/core/src/common/message-rpc/rpc-message-encoder.ts

```diff
@@ -388,6 +428,16 @@ export class RpcMessageEncoder {
             write: (buf, value) => buf.writeString(JSON.stringify(value))
         });
 
+        this.registerEncoder(ObjectType.Map, {
+            is: value => value instanceof Map,
```

Comment:
Don't we have to treat anything that can hold a reference the same as the `typeof x === object`? I mean I can do

```
const x= new Set();
const y= new Set();
x.add(y);
y.add(x);
```
Bingo, circular reference 🤷

Reply:
`Set`s aren't a problem here, but JavaScript is:

```javascript
const a = new Set();
// undefined
typeof a
// 'object'
const b = 'hello'
// undefined
const c = 'other string'
// undefined
 b.c = c;
// 'other string'
 c.b = b
// 'hello'
typeof b
// 'string'

Reply:
Good catch. I have refactored the code and now all object types (other than null/undefined) are going through the reference check. I have also added a test case for circular references based on  the example that @tsmaeder provided.